### PR TITLE
RFC: refactor copy argument sanitizing into a unified compat function

### PR DESCRIPTION
**Pull request by neutrinoceros**
### Description
This is a follow up to #16142 so we have a single source of truth regarding how we treat user-provided `copy=False` arguments.

This is also a step forward followin my proposed long term plan (see #16167), but it's in itself not an API cchange and could be backported for convenience if desired.

- [ ] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

File: astropy/utils/compat/numpycompat.py [hunk 1]
```diff
@@ -26,3 +27,12 @@
 
 
 COPY_IF_NEEDED = False if NUMPY_LT_2_0 else None
+
+
+def sanitize_copy_arg(copy, /):
+    if not NUMPY_LT_2_0 and copy is False:
+        # TODO: deprecate this path, and eventually remove this entire function
```
Comment (neutrinoceros) [inline astropy/utils/compat/numpycompat.py:34]: this TODO could be updated or removed. It isn't needed for this PR, and largely depends on future conversation in #16167

**Review by pllim — COMMENTED** [review]
I thought we agreed to just follow numpy 2.x convention if people have numpy 2.x, and if people want the old numpy 1.x behavior, they should just downgrade numpy back to 1.x. With that in mind, this PR is completely unnecessary?

**Comment by neutrinoceros** [comment]
This PR is actually older than the discussion, but I think it still has value in that it prevents that some paths are updated and not others. I could just remove or update the TODO comment to avoid making promises on the future.

**Comment by mhvk** [comment]
Is this still needed? The conclusion to just live with whatever numpy does suggests that all we need to do is update any `copy=False` defaults to `copy=COPY_IF_NEEDED` (with the old `COPY_IF_NEEDED = False if NUMPY_LT_2_0 else None`) and that otherwise code can remain untouched. Or am I missing something?

**Comment by neutrinoceros** [comment]
That  was indeed a clear consensus amongst participants in the last dev telecon, however we also agreed that we should ultimately involve downstream stake holders and give them a chance to test different approaches so we can settle for the most aggressive one that doesn't cause too much immediate pain. With this in mind, there are two alternative continuations to this PR:
- #16174
- #16181

the second one just removes the function I'm introducing here, but the first one creates a deprecation path instead. So, if it turns out downstream players are not affected by the API change, we'll indeed just end up removing this function. However, given the uncertainty, I believe there is still value in having one source of truth instead of multiple instances of this code bits.
I would also argue that it could actually be more work to *not* merge this one since I'd need to slightly refactor #16174 and #16181 . Nothing too bad, but from where I stand this one is just a button push away from done.

**Review by mhvk — APPROVED** [review]
Ah, I see, I missed that c a deprecation warning would indeed much more easily raised inside a function. Since you're all set up to do this, and we will either end up removing this or adding a deprecation warning, I think this is good to go. Thanks, @neutrinoceros!

**Comment by mhvk** [comment]
p.s. Not sure what's up with the tests that just seem to be waiting to be run.

**Comment by neutrinoceros** [comment]
I think that's a side effect of merging #15603. If it's blocking I can rebase this one.

**Comment by mhvk** [comment]
I cannot override the statuses, so, yes, maybe best to rebase so we can actually merge...

**Comment by neutrinoceros** [comment]
done

**Comment by neutrinoceros** [comment]
Oh, actually blocked by https://github.com/scientific-python/pytest-doctestplus/pull/242

**Comment by neutrinoceros** [comment]
closing/reopening to retrigger CI now that pytest-doctestplus was updated

File: astropy/utils/compat/numpycompat.py (hunk 1 above)
Comment (neutrinoceros) [reply astropy/utils/compat/numpycompat.py]: (I removed it)

**Comment by mhvk** [comment]
OK, thanks, let's get this in so the other two can be rebased...

**Comment by neutrinoceros** [comment]
Thanks Marten ! every child branches are now rebased and conflict-free

**Pull request merged**